### PR TITLE
SVS-132: Fail to bind (connected mode) when solution contains a VB.Net project but the server doesn't have the plugin

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: src/Integration/MefServices/IHost.cs [hunk 1]
```diff
@@ -60,5 +61,7 @@ internal interface IHost : IServiceProvider
         /// Clears the <see cref="ActiveSection"/>
         /// </summary>
         void ClearActiveSection();
+
+        ISet<Language> SupportedPluginLanguages { get; set; }
```
Comment (Evangelink) [inline src/Integration/MefServices/IHost.cs]: Not the biggest fan of this solution but I haven't found any nicer solution.

**Comment by tvsonar** [comment]
@Evangelink It's quite difficult to review this PR due to the formatting changes. :-(

File: src/Integration/Binding/BindingWorkflow.cs [hunk 2]
```diff
@@ -439,22 +435,15 @@ private void InformAboutFilteredOutProjects()
...
-            }
-
-            output = string.Format(Strings.SubTextPaddingFormat, output);
-
-            VsShellUtils.WriteToSonarLintOutputPane(this.host, output);
+            var output =
+                profile == null
+                    ? string.Format(Strings.CannotDownloadQualityProfileForLanguage, languageName)
+                    : isDownloadFailed
+                        ? string.Format(Strings.QualityProfileDownloadFailedMessageFormat, profile.Name, profile.Key, languageName)
```
Comment (tvsonar) [inline src/Integration/Binding/BindingWorkflow.cs]: You can merge these two `string.Format`s by assigning the format to a local variable.  Also, nested `?:`s are hard to read, let's put the second one in a dedicated method.

File: src/Integration/Connection/ConnectionWorkflow.cs [hunk 3]
```diff
@@ -187,64 +188,72 @@ private void OnProjectsChanged(ConnectionInformation connection, ProjectInformat
...
-            {
-                errorMessage = string.Format(Strings.OnlySupportedPluginHasNoProjectInSolution, Language.CSharp.Name);
-            }
-            else
-            {
-                errorMessage = string.Format(Strings.OnlySupportedPluginHasNoProjectInSolution, Language.VBNET.Name);
-            }
 
-            this.host.ActiveSection?.UserNotifications?.ShowNotificationError(errorMessage, NotificationIds.BadServerPluginId, null);
+            string errorMessage = GetPluginProjectMismatchErrorMessage(csharpOrVbNetProjects);
```
Comment (tvsonar) [inline src/Integration/Connection/ConnectionWorkflow.cs]: This refactored version is a lot easier to read then the original one. +1

File: src/Integration/Binding/BindingWorkflow.cs [hunk 4]
```diff
@@ -393,7 +387,9 @@ private void AbortWorkflow(IProgressController controller, CancellationToken tok
 
         internal /*testing purposes*/ IEnumerable<Language> GetBindingLanguages()
         {
-            return this.BindingProjects.Select(Language.ForProject).Distinct();
+            return this.BindingProjects.Select(Language.ForProject)
+                                       .Distinct()
+                                       .Where(l => this.host.SupportedPluginLanguages.Contains(l));
```
Comment (tvsonar) [inline src/Integration/Binding/BindingWorkflow.cs]: `l` doesn't mean anything. Let's get rid of it with a method group, or let's rename it to `language`.

**Pull request merged**